docs/api/v2/node.md

```diff
@@ -25,14 +25,11 @@ method subscribe*(w: WakuNode, topic: Topic, handler: TopicHandler)
   ## NOTE The data field SHOULD be decoded as a WakuMessage.
   ## Status: Implemented.
 
-method subscribe*(w: WakuNode, contentFilter: ContentFilter, handler: ContentFilterHandler)
-  ## Subscribes to a ContentFilter. Triggers handler when receiving messages on
-  ## this content filter. ContentFilter is a method that takes some content
-  ## filter, specifically with `ContentTopic`, and a `Message`. The `Message`
-  ## has to match the `ContentTopic`.
-
-  ## Status: Not yet implemented.
-  ## TODO Implement as wrapper around `waku_filter` and `subscribe` above.
+proc filter*(w: WakuNode, filter: FilterRequest, handler: FilterHandler) {.async, gcsafe.} =
```

Comment:
I'm not a huge fan of using filter as a name. Publish and Subscribe are clear verbs, and Filter is overused:
- In FP with map/filter/reduce which has different semantics (not a "listen" function)
-  In Waku v1 with different semantics
(- and IIRC previously in nim-waku with "filters pubsub" implementation detail, though I think this is notify now)

That is bound to lead to confusion and poor mental models IMO.

The argument to not use `subscribe` here is that it is overloaded. Personally I don't see the issue with this, and it seems quite idiomatic in Nim code base from what I can tell (@kdeme might know better). That said, I'd be OK changing it if people feel strongly about it. So if we don't want to use `subscribe` here, how about `subscribeContent` or `subscribeWithFilter`? Or possibly `filterSubscribe`

Reply:
Also, why are we changing the name of ContentFilterHandler to FilterHandler? The former is more clear to me in terms of intent. Don't see any advantages to this change except saving a few characters.

Reply:
@oskarth the ContentFilterHandler is something completely different and seems to be used elsewhere. Its signature is of `proc(message: seq[byte])`

Reply:
Discussed in PM, this is inaccurate. It is just an old version of the what it looks like now. They refer to the same thing, or should.

Reply:
yup, will change. Thanks for clarification.